prog/vnet/ubicloud/subnet_nexus.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::Ubicloud::SubnetNexus < Prog::Base
+  subject_is :private_subnet
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        register_deadline(nil, 10 * 60)
+        hop_destroy
+      end
+    end
+  end
+
+  label def start
+    hop_wait
+  end
+
+  label def wait
+    when_refresh_keys_set? do
+      private_subnet.update(state: "refreshing_keys")
+      hop_refresh_keys
+    end
+
+    when_add_new_nic_set? do
+      private_subnet.update(state: "adding_new_nic")
+      hop_add_new_nic
+    end
+
+    when_update_firewall_rules_set? do
+      private_subnet.vms.each(&:incr_update_firewall_rules)
+      decr_update_firewall_rules
+    end
+
+    if private_subnet.last_rekey_at < Time.now - 60 * 60 * 24
+      private_subnet.incr_refresh_keys
+    end
+
+    nap 10 * 60
+  end
+
+  def gen_encryption_key
+    "0x" + SecureRandom.bytes(36).unpack1("H*")
+  end
+
+  def gen_spi
+    "0x" + SecureRandom.bytes(4).unpack1("H*")
+  end
+
+  def gen_reqid
+    SecureRandom.random_number(100000) + 1
```

Comment:
If this should only return a non-zero 1-5 digit number, this has a problem if the random number is `99999`. It may be better to do:

```suggestion
    SecureRandom.random_number(1...100000)
```

If this should be allowed to return a non-zero 1-6 digit number, then you should use the above and add a zero.

Reply:
I did this change, but it seems it is lost during one of the rebases. Weird... I fixed it again.

Reply:
I mistakenly did it  in the original subnet nexus prog. Fixed it now.